addons/stock/product.py

```diff
@@ -374,11 +382,19 @@ def _product_available_text(self, cr, uid, ids, field_names=None, arg=False, con
         return res
 
     def _compute_nbr_reordering_rules(self, cr, uid, ids, field_names=None, arg=None, context=None):
-        res = dict.fromkeys(ids, 0)
-        product_data = self.pool['stock.warehouse.orderpoint'].read_group(cr, uid, [('product_id.product_tmpl_id', 'in', ids)], ['product_id'], ['product_id'], context=context)
+        res = dict.fromkeys(ids, {'nbr_reordering_rules': 0, 'reordering_min_qty': 0, 'reordering_max_qty': 0})
+        product_data = self.pool['stock.warehouse.orderpoint'].read_group(cr, uid, [('product_id.product_tmpl_id', 'in', ids)], ['product_id', 'product_min_qty', 'product_max_qty'], ['product_id'], context=context)
+
+        for data in product_data:
+            product_tmpl_id = data['__domain'][1][2][0]
+            res[product_tmpl_id]['nbr_reordering_rules'] += data['product_id_count']
+
         for data in product_data:
             product_tmpl_id = data['__domain'][1][2][0]
-            res[product_tmpl_id] += data['product_id_count']
+            if res[product_tmpl_id]['nbr_reordering_rules'] == 1:
+                res[product_tmpl_id]['reordering_min_qty'] += data['product_min_qty']
+                res[product_tmpl_id]['reordering_max_qty'] += data['product_max_qty']
+
```

Comment:
regarding two loops: i think this is due to master code. some of code for reording rules count already merged in master. i think this branch required rebase.